build_options.cmake

```diff
@@ -52,8 +54,27 @@ function( TNUN_add_compile_options configuration )
     endforeach()
 endfunction()
 
+TNUN_add_compile_options( Debug ${TNUN_compiler_debug_flags} ${TNUN_compiler_debug_symbols} ${TNUN_compiler_runtime_sanity_checks} )
 TNUN_add_compile_options( Release ${TNUN_compiler_release_flags} )
 
+option( TNUN_DEBUG_SYMBOLS_IN_RELEASE "Generate debug symbols for easier debugging of release builds" false )
+if ( ${TNUN_DEBUG_SYMBOLS_IN_RELEASE} )
+    TNUN_add_compile_options( Release ${TNUN_compiler_debug_symbols} )
+endif()
+
+option( TNUN_ALLOW_EXCEPTIONS "Allow exception support in C++ code" true )
+if( ${TNUN_ALLOW_EXCEPTIONS} )
+    add_compile_options( ${TNUN_compiler_exceptions_on} )
+else()
+    add_compile_options( ${TNUN_compiler_exceptions_off} )
+endif()
+
+option( TNUN_ALLOW_RTTI "Allow Runtime Type Information in C++ code" true )
+if( ${TNUN_ALLOW_RTTI} )
+    add_compile_options( ${TNUN_compiler_rtti_on} )
+else()
+    add_compile_options( ${TNUN_compiler_rtti_off} )
+endif()
```

Comment:
not to happy about this block: in ide's/multiconfig generators you'll often want rtti in debug but not in release and also, although that is less common, the same for exceptions..